File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}
        for n in nums:
            freq[n] = freq.get(n, 0) + 1
        heap = [[f, v] for v, f in freq.items()]
        heapq.heapify(heap)
        while len(heap) > k:
            heapq.heappop(heap)
        res = [v for f, v in heap]
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-4.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        HashSet<int> res = new HashSet<int>();
        for (int i = 0; i < nums.Length; i++){
            if (res.Contains(nums[i])) {
                return true;
            }
            res.Add(nums[i]);
        }
        return false;
    }
}